node/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -269,18 +269,15 @@ getWalletAddresses =
     view wsWalletInfos
 
 getAccountWAddresses :: AddressLookupMode
-                  -> AccountId
-                  -> Query (Maybe [CWAddressMeta])
+                     -> AccountId
+                     -> Query (Maybe [AddressInfo])
 getAccountWAddresses mode accId =
     withAccLookupMode mode (fetch aiAddresses) (fetch aiRemovedAddresses)
   where
-    fetch :: MonadReader WalletStorage m => Lens' AccountInfo CAddresses -> m (Maybe [CWAddressMeta])
+    fetch :: MonadReader WalletStorage m => Lens' AccountInfo CAddresses -> m (Maybe [AddressInfo])
     fetch which = do
-        cAddresses <- preview (wsAccountInfos . ix accId . which)
-        -- here `cAddresses` has type `Maybe CAddresses`
-        pure $
-            (map adiCWAddressMeta . sortOn adiSortingKey . map snd . HM.toList)
-            <$> cAddresses
+        addresses <- preview (wsAccountInfos . ix accId . which)
+        pure (HM.elems <$> addresses)
```

Comment:
This can now be simpler still:

    fetch which = HM.elems <$> preview (wsAccountInfos . ix accId . which)

Reply:
@dcoutts Done 😉